.github/workflow/ci.yml

```diff
@@ -0,0 +1,45 @@
+---
+# This is a basic workflow to help you get started with Actions
+name: CI
+
+# Controls when the workflow will run
+on:
+  # Triggers the workflow on push or pull request
+  # events but only for the main branch
+  push:
+    branches: [main]
+    paths-ignore:
+      - '**/README.md'
+  pull_request:
+    branches: [main]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  test:
+    name: Test on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
```

Comment:
If you are only running flutter analyze and only building the app on linux, then I don't think all platforms are needed as they all will report the same.

Reply:
I'll make a test on Adwaita to see if we could build the app for all platforms. If not, I'll remove the matrix test.